src/mainviews/job-definition.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useMemo } from 'react';
+import { IJobDefinitionModel, JobsView } from '../model';
+import { useTranslator } from '../hooks';
+import { TextFieldStyled, timestampLocalize } from './job-detail';
+import { SchedulerService } from '../handler';
+import cronstrue from 'cronstrue';
+
+import {
+  Button,
+  Card,
+  CardContent,
+  Stack,
+  TextFieldProps
+} from '@mui/material';
+
+export interface IJobDefinitionProps {
+  model: IJobDefinitionModel;
+  refresh: () => void;
+  setView: (view: JobsView) => void;
+}
+
+export function JobDefinition(props: IJobDefinitionProps): JSX.Element {
+  const trans = useTranslator('jupyterlab');
+
+  const ss = useMemo(() => new SchedulerService({}), []);
+
+  const handleDeleteJobDefinition = async () => {
+    await ss.deleteJob(props.model.definitionId ?? '');
+    props.setView('ListJobs');
```

Comment:
Shouldn't we setting the view to "Job definition list" in this case? The create form now does this, for example, when you create a job definition.

Reply:
What's confusing is that `model.jobsView = 'ListJobs'` will render either the job list or the job definition list, dispatching based on `model.listJobsModel.listJobsView`.

@andrii-i For now, you'll want to somehow set the `listJobsView` to `'JobDefinitionDetail'` immediately after line 29 above.

@jweill-aws We need to have a more thorough discussion on the purpose of these models. To my recollection, the whole point of models as a global state store was to allow persistence (serialization to JSON in local storage) and rehydration upon page reload. However, the list jobs view really doesn't have any state that needs to be persisted. Whether the jobs list or the job definitions list is being rendered should be stored in `model.jobsView` rather than inside `model.listJobsModel.listJobsView`, which means `model.listJobsModel` can be deleted entirely.

Another refactoring suggestion, rename `JobsModel` => `SchedulerModel` and `model.jobsView` => `model.view`.

Reply:
I also want to discuss getting rid of `model.jobDetailModel.detailType`, and store all "view" information inside of `model.jobsView`/`model.view`. @jweill-aws This will probably be a much larger discussion than anticipated, so let's address model refactoring in a future PR.

Reply:
Created #105 to track this

Reply:
setting listJobsView to 'JobDefinition' ('JobDefinitionDetail' was a redundant JobsView option, removed)